src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -18,9 +18,26 @@
 using Microsoft.VisualStudio.Workspace;
 using Microsoft.VisualStudio.Workspace.Indexing;
 using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+using Microsoft.Win32;
 
 namespace Microsoft.DebugEngineHost
 {
+    internal class RegisterMonitorWrapper : IDisposable
```

Comment:
>RegisterMonitorWrapper [](http://example.com/codeflow?start=19&length=22)

Maybe I will eventually figure this out, but why do we need a wrapper instead of just directly holding the registry monitor?

Reply:
I think this was before I discovered the MIDebugEngine did not really need to know about the RegistorMonitor class and we could just cover cleanup with IDisposable.

Reply:
Nope just remembered why myself now 😅

It was because of `WatchNatvisOptionSetting` and we need to switch out the registry we are watching since we are watching two different registry keys.